lib/Locations.php

```diff
@@ -65,4 +75,18 @@ public function first(): Location
 
         return reset($this->locations);
     }
+
+    public function sorted(): self
+    {
+        usort($this->locations, function (Location $first, Location $second) {
+            $order = strcmp((string) $first->uri(), (string) $second->uri());
+            if (0 !== $order) {
+                return $order;
+            }
+
+            return $first->offset()->toInt() - $second->offset()->toInt();
+        });
+
+        return $this;
```

Comment:
this needs to return a new instance. the locations should be immuatble.

```
$locations = $this->locations;
usort(...);
return new self($locations);
```

Reply:
My bad, just saw the `append()` method but didn't check the implementation.
I'm used to `withXXX()` method for immutable objects ><